crates/db_schema/src/lib.rs

```diff
@@ -157,6 +157,29 @@ pub enum PostListingMode {
   SmallCard,
 }
 
+#[derive(EnumString, Display, Debug, Serialize, Deserialize, Clone, Copy, PartialEq, Eq, Hash)]
+#[cfg_attr(feature = "full", derive(DbEnum, TS))]
+#[cfg_attr(
+  feature = "full",
+  ExistingTypePath = "crate::schema::sql_types::VoteDisplayModeEnum"
+)]
+#[cfg_attr(feature = "full", DbValueStyle = "verbatim")]
+#[cfg_attr(feature = "full", ts(export))]
+// TODO, in future releases, remove hide_scores, as this supercedes it.
+/// A vote-display setting that changes how votes are displayed in front ends.
+pub enum VoteDisplayMode {
+  /// Shows the score, upvotes, and downvotes.
+  Full,
+  /// Shows the score, and upvote percentage. (the default setting)
+  ScoreAndUpvotePercentage,
+  /// Shows the upvote percentage only.
+  UpvotePercentage,
+  /// Shows the score, but hides downvotes.
+  HideDownvotes,
```

Comment:
Showing the score and hiding just the downvotes is functionally the same as only just showing the total score with both upvotes and downvotes, since users would be able get the downvote number by subtracting the number of upvotes from the total score. With that in mind, I think having this as just `Score` is better.

Reply:
HideDownvotes also wouldn't work on instances that have downvotes disabled

Reply:
Makes sense. I'll add the downvote stuff as a comment.